src/main/java/org/alliancegenome/curation_api/services/validation/dto/fms/ExternalDataBaseEntityFmsDTOValidator.java

```diff
@@ -0,0 +1,146 @@
+
+	@Transactional
+	public ExternalDataBaseEntity validateExternalDataBaseEntityFmsDTO(HTPIdFmsDTO dto) throws ObjectValidationException {
+
+		ObjectResponse<ExternalDataBaseEntity> externalDBEntityResponse = new ObjectResponse<>();
+		ExternalDataBaseEntity externalDBEntity = null;
+		
+		if(StringUtils.isEmpty(dto.getPrimaryId())) {
+			externalDBEntityResponse.addErrorMessage("primaryId", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			externalDBEntity = externalDataBaseEntityService.findByCurie(dto.getPrimaryId());
+		}
+
+		if(externalDBEntity == null) {
+			externalDBEntity = new ExternalDataBaseEntity();
+			externalDBEntity.setCurie(dto.getPrimaryId());
+		}
+
+		if(CollectionUtils.isNotEmpty(dto.getAlternateIds())) {
+			if(externalDBEntity.getSecondaryIdentifiers() == null) {
+				externalDBEntity.setSecondaryIdentifiers(new ArrayList<>());
+			}
+			Set<String> existingSecondaryIds = new HashSet<>(externalDBEntity.getSecondaryIdentifiers());
+			Set<String> incomingAlternateIds = new HashSet<>(dto.getAlternateIds());
+
+			existingSecondaryIds.retainAll(incomingAlternateIds);
+			existingSecondaryIds.addAll(incomingAlternateIds);
+			externalDBEntity.setSecondaryIdentifiers(new ArrayList<>(existingSecondaryIds));
+		}
+
+		List<Long> currentXrefIds;
+		if (externalDBEntity.getCrossReferences() == null) {
+			currentXrefIds = new ArrayList<>();
+		} else {
+			currentXrefIds = externalDBEntity.getCrossReferences().stream().map(CrossReference::getId).collect(Collectors.toList());
+		}
+
+		List<Long> mergedXrefIds = null;
+		List<CrossReference> mergedCrossReferences = new ArrayList<>();
+		if(CollectionUtils.isNotEmpty(dto.getCrossReferences())) {
+			mergedCrossReferences = crossReferenceService.getMergedFmsXrefList(dto.getCrossReferences(), externalDBEntity.getCrossReferences());
+			mergedXrefIds = mergedCrossReferences.stream().map(CrossReference::getId).collect(Collectors.toList());
+			externalDBEntity.setCrossReferences(mergedCrossReferences);
+		} else {
+			mergedXrefIds = new ArrayList<>();
+			externalDBEntity.setCrossReferences(null);
+		}
+
+		Map<String, CrossReference> mergedXrefUniqueIdsMap = new HashMap<>();
+		for(CrossReference mergedXref : mergedCrossReferences) {
+			mergedXrefUniqueIdsMap.put(crossReferenceService.getCrossReferenceUniqueId(mergedXref), mergedXref);
+		}
+
+		if(externalDBEntity.getPreferredCrossReference() != null) {
+			if(dto.getPreferredCrossReference() != null) {
+				CrossReference incomingPreferredXref = createNewCrossReference(dto.getPreferredCrossReference());
+				String incomingXrefUniqueId = crossReferenceService.getCrossReferenceUniqueId(incomingPreferredXref);
+				String currentXrefUniqueId = crossReferenceService.getCrossReferenceUniqueId(externalDBEntity.getPreferredCrossReference());
+
+				if(!incomingXrefUniqueId.equals(currentXrefUniqueId)){
+					externalDBEntity.setPreferredCrossReference(null);
+					if(mergedXrefUniqueIdsMap.containsKey(incomingXrefUniqueId)) {
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' is not followed by whitespace.
<!-- __reviewdog__:ChA0MWZmNzRlNTI0Zjk0MjU2EgpjaGVja3N0eWxl -->